libraries/ESP8266WiFiMesh/examples/HelloMesh/HelloMesh.ino

```diff
@@ -1,50 +1,129 @@
+  Serial.println(response);
+
+  // Our last request got a response, so time to create a new request.
+  mesh_instance->setMessage("Hello world request #" + String(++request_i) + " from " + mesh_instance->getMeshName() + mesh_instance->getNodeID() + ".");
+
+  // (void)mesh_instance; // This is useful to remove a "unused parameter" compiler warning. Does nothing else.
+  return status_code;
+}
+
 void setup() {
+  // Prevents the flash memory from being worn out, see: https://github.com/esp8266/Arduino/issues/1054 .
+  // This will however delay node WiFi start-up by about 700 ms. The delay is 900 ms if we otherwise would have stored the WiFi network we want to connect to.
+  WiFi.persistent(false);
+
   Serial.begin(115200);
-  delay(10);
+  delay(50); // Wait for Serial.
+
+  //yield(); // Use this if you don't want to wait for Serial.
 
   Serial.println();
   Serial.println();
+
+  Serial.println("Note that this library can use static IP:s for the nodes to speed up connection times.\n"
+                 "Use the setStaticIP method to enable this and ensure that nodes connecting to the same AP have distinct static IP:s.\n"
+                 "Also, remember to change the default mesh network password!\n\n");
+
   Serial.println("Setting up mesh node...");
 
   /* Initialise the mesh node */
   mesh_node.begin();
+  mesh_node.activateAP(); // Each AP requires a separate server port.
+  mesh_node.setStaticIP(IPAddress(192, 168, 4, 22)); // Activate static IP mode to speed up connection times.
 }
 
+int32_t time_of_last_scan = -10000;
 void loop() {
-  /* Accept any incoming connections */
-  mesh_node.acceptRequest();
-
-  /* Scan for other nodes and send them a message */
-  char request[60];
-  sprintf(request, "Hello world request #%d from Mesh_Node%d.", request_i++, ESP.getChipId());
-  mesh_node.attemptScan(request);
-  delay(1000);
+  if (millis() - time_of_last_scan > 3000 // Give other nodes some time to connect between data transfers.
+      || (WiFi.status() != WL_CONNECTED && millis() - time_of_last_scan > 2000)) { // Scan for networks with two second intervals when not already connected.
+    char request[60];
+    sprintf(request, "Hello world request #%d from %s%s.", request_i, mesh_node.getMeshName().c_str(), mesh_node.getNodeID().c_str());
+    mesh_node.attemptTransmission(request, false);
+    time_of_last_scan = millis();
+
+    if (ESP8266WiFiMesh::latest_transmission_outcomes.empty()) {
+      Serial.println("No mesh AP found.");
+    } else {
+      for (TransmissionResult &transmission_result : ESP8266WiFiMesh::latest_transmission_outcomes) {
+        if (transmission_result.transmission_status == TS_TRANSMISSION_FAILED) {
+          Serial.println("Transmission failed to mesh AP " + transmission_result.ssid);
+        } else if (transmission_result.transmission_status == TS_CONNECTION_FAILED) {
+          Serial.println("Connection failed to mesh AP " + transmission_result.ssid);
+        }
```

Comment:
else ? (Robustness)

Reply:
In this particular case I’d argue that we only care about if there is an error occurring, and those are the cases checked for here. Adding an else would not really aid the user since it would only capture all the successful transmissions which we already get confirmations about elsewhere in the code (from the responseHandler).

Reply:
If there is ever another error status value added and returned, the if-else-if clause will silently not catch anything. For cases like this, the else should be akin to an assert. It's a similar case as having default: in a switch. It is rarely a good practice not to have one.

Reply:
I realized that a good way to get the robustness advantage you mentioned while avoiding too many notification events for the user was simply to add a condition for every current state, and then an else. This is now implemented in the most recent code update.